lib/takeaway.rb

```diff
@@ -0,0 +1,84 @@
+require 'twilio-ruby'
+
+class TakeAway
+  SECRET_KEY = ENV['SECRET_KEY']
+
+  attr_reader :menu
+  attr_accessor :order
+
+  def initialize
+    @order = {}
+  end
+
+  def add_menu(menu)
+    @menu = menu.dishes
+  end
+
+  def place_order(items = {})
+    items.each do |dish, quantity|
+      raise "#{dish.capitalize} is not on the menu." unless menu.include?(dish)
+      item_already_ordered?(dish) ? change_item_quantity(dish, quantity) : order["#{dish}"] = quantity
+    end
+  end
+
+  def item_already_ordered?(item)
+    true if order.include?(item)
+  end
+
+  def change_item_quantity(item, quantity)
+    current_quantity = order[item]
+    order[item] = current_quantity + quantity
+  end
+
+  def grand_total
+    total_in_pence = 0
+    order.each { |item, quantity| total_in_pence += total_cost_per_item(item, quantity) }
+    "%.2f" % (total_in_pence / 100.0)
+  end
+
+  def total_cost_per_item(item, quantity)
+    price_in_pence = ((menu[item] * 100).round * quantity)
+  end
+
+  def show_order
+    order_output = print_order_items_with_cost
+    order_output += "Total: £#{grand_total}"
+  end
+
+  def delete_from_order(item, quantity)
+    new_quantity = order[item] - quantity
+    order.delete("#{item}")
+    order[item] = new_quantity if new_quantity > 0
+  end
+
+  def confirm_order(phone_number)
+    puts "Please confirm that your order is correct by typing Y for Yes or N for No."
```

Comment:
Line is too long. [85/80]<br>Do not write to stdout. Use Rails' logger if you want to log.